io.github.bkbilly.lnxlink.yaml

```diff
@@ -0,0 +1,50 @@
+id: io.github.bkbilly.lnxlink
+runtime: org.gnome.Platform
+runtime-version: '46'
+sdk: org.gnome.Sdk
+command: lnxlink
+modules:
+  - flatpak-modules-meson.yaml
+  - flatpak-modules-scikit.yaml
+  - flatpak-modules.yaml
+  - flatpak-modules-opencv.yaml
+  - flatpak-lnxlink.yaml
+  - flatpak-modules-system.json
+  - name: metadata
+    buildsystem: simple
+    build-commands:
+      - install -Dm644 -t ${FLATPAK_DEST}/share/metainfo ${FLATPAK_ID}.metainfo.xml
+    sources:
+      - type: file
+        path: io.github.bkbilly.lnxlink.metainfo.xml
+finish-args:
+  # Wayland access
+  - --share=ipc
+  - --socket=wayland
+  - --socket=fallback-x11
+  # DBus
+  # - --socket=system-bus
+  # - --socket=session-bus
+  # For DBus
+  - --system-talk-name=org.freedesktop.NetworkManager
+  - --system-talk-name=org.freedesktop.login1
+  - --system-talk-name=org.freedesktop.UPower
+  - --talk-name=org.freedesktop.Notifications
+  - --talk-name=org.gnome.Mutter.IdleMonitor
+  - --talk-name=org.mpris.MediaPlayer2.*
+  # Access to network
+  - --share=network
+  # Access to audio
+  - --socket=pulseaudio
+  # Access to Bluetooth
+  - --allow=bluetooth
+  # Needs to save files locally
+  - --filesystem=xdg-documents
+  # Access to filesystem
+  - --filesystem=/tmp/
+  - --filesystem=/var/run/reboot-required
```

Comment:
why do you need this?

Reply:
The /tmp path is required for the notification image: https://github.com/bkbilly/lnxlink/blob/master/lnxlink/modules/notify.py

The reboot-required is for checking if the system needs to be rebooted: https://github.com/bkbilly/lnxlink/blob/master/lnxlink/modules/required_restart.py

Reply:
Both of them are needed for my application to work properly.

Reply:
Don't hardcode `tmp` use the tempfile library https://docs.python.org/3/library/tempfile.html

Then set TMPDIR in the finish-args.

Reply:
don't hardcode /tmp like that. Use the tempfile library, then set TMPDIR in the manifest.

Reply:
How should I set the TMPDIR?

Reply:
`--env=TMPDIR=value` if you want persistent use `/var/tmp` for non-persistent `${XDG_RUNTIME_DIR}/app/${FLATPAK_ID}`

Reply:
I still can't understand how to use it.
Under the finish-args I've removed the `filesystem=/tmp/` and added the TMPDIR environment but it doesn't have access to the `/tmp` filesystem:
```yaml
- --env=TMPDIR=/tmp/
```

I've also tried adding the variable you suggested, but this time it doesn't even build:
```yaml
- --filesystem=${XDG_RUNTIME_DIR}/app/${FLATPAK_ID}
```

I am using the tempfile library which defaults to the `/tmp/` folder, not the `/var/tmp/`.

Reply:
You don't need `/tmp`, the directory used in TMPDIR will be your new temp folder.

https://docs.python.org/3/library/tempfile.html

> The default directory is chosen from a platform-dependent list, but the user of the application can control the directory location by setting the TMPDIR, TEMP or TMP environment variables.****

Reply:
My application creates temporary images that they need to be deleted by the OS when it restarts, but as long as the OS is running they need to exist because they are used by the system notification. 
The user has the option to open it at any time, as long as the notification is active via action buttons. This is why I set the tempfile with these options:
```python
tempfile.NamedTemporaryFile(prefix="lnxlink_icon_", delete=False, delete_on_close=False)
```

It seems to me that setting the TMPDIR won't work for the requirements of my application.

Reply:
Then what's the issue with setting `--env=TMPDIR=${XDG_RUNTIME_DIR}/app/${FLATPAK_ID}` ?

Reply:
Also why does notification images need to be temporary files?

Reply:
Yea, right missed that.

Reply:
The notification needs temporary files for the images and sounds. These are downloaded to a temporary file and they are called by the notification system or can be shown as full screen when the user clicks on a provided button. 
The user can click on that button at any point that the OS is turned on, so they need to exist and be deleted once it restarts because otherwise it will have a lot of unwanted files.
This image could be used for any application that the user wants by creating automations on Home Assistant. For example it could be used to be sent as an email attachment or anything else.

Reply:
I'm guessing notifications here mean desktop notifications? So these icons need to be accessible outside the sandbox for the notification daemon on host to show the icon?

Reply:
Yeah, that's how the icons are used.

Reply:
I tried that, but unfortunately it doesn't work because when the application starts, the directory `/tmp/lnxlink` doesn't exist so I create it.
```python
if not os.path.exists("/tmp/lnxlink"):
    os.makedirs("/tmp/lnxlink")
```
Once it is created with flatpak, the system can't see the newly created directory.
In order to work the way you suggest, the directory `/tmp/lnxlink` has to exist before running the application.

Reply:
The `:create` prefix exists for that. `--filesystem=/tmp/lnxlink:create` will create the directory if it doesn't exist first.

Reply:
This works perfectly! Thanks for your suggestion!